chalk-solve/src/clauses/program_clauses.rs

```diff
@@ -402,28 +402,17 @@ impl<I: Interner> ToProgramClauses<I> for AdtDatum<I> {
                 ($goal:ident) => {
                     // Fundamental types must always have at least one
                     // type parameter for this rule to make any
-                    // sense. We currently do not have have any
-                    // fundamental types with more than one type
-                    // parameter, nor do we know what the behaviour
-                    // for that should be. Thus, we are asserting here
-                    // that there is only a single type parameter
-                    // until the day when someone makes a decision
-                    // about how that should behave.
-                    assert_eq!(
-                        self_appl_ty.len_type_parameters(interner),
-                        1,
-                        "Only fundamental types with a single parameter are supported"
-                    );
-
-                    builder.push_clause(
-                        DomainGoal::$goal(self_ty.clone()),
-                        Some(DomainGoal::$goal(
-                            // This unwrap is safe because we asserted
-                            // above for the presence of a type
-                            // parameter
-                            self_appl_ty.first_type_parameter(interner).unwrap(),
-                        )),
+                    // sense.
+                    assert!(
```

Comment:
Maybe this is preexisting. But more applicable now that we only use it twice. But I feel like the macro isn't super necessary. We can do an assert if fundamental. And just write the goal out in the two places. Would also reduce the asserts from two to one, though I'm not super worried about that.